extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosImporter.java

```diff
@@ -142,38 +142,57 @@ private void importSinglePhoto(UUID id, PhotoModel photo) throws FlickrException
     if (Strings.isNullOrEmpty(oldAlbumId)) {
       return;
     }
+    createOrAddToAlbum(idempotentExecutor, id, photo.getAlbumId(), photoId);
+  }
 
-    TempPhotosData tempData = jobStore.findData(id, createCacheKey(), TempPhotosData.class);
-    String newAlbumId = tempData.lookupNewAlbumId(oldAlbumId);
+  private void createOrAddToAlbum(
+      IdempotentImportExecutor idempotentExecutor,
+      UUID jobId,
+      String oldAlbumId,
+      String photoId) throws IOException, FlickrException {
+    String newAlbumId = jobStore.findData(jobId, oldAlbumId, String.class);
 
     if (Strings.isNullOrEmpty(newAlbumId)) {
-      // This means that we havent created the new album yet, create the photoset
-      PhotoAlbum album = tempData.lookupTempAlbum(oldAlbumId);
-
-      // TODO: handle what happens if the album doesn't exist. One of the things we can do here is
-      // throw them into a default album or add a finalize() step in the Importer which can deal
-      // with these (in case the album exists later).
-      Preconditions.checkArgument(album != null, "Album not found: " + oldAlbumId);
-
-      // TODO: do we want to keep the COPY_PREFIX?  I feel like not
-      String albumName =
-          Strings.isNullOrEmpty(album.getName()) ? "" : COPY_PREFIX + album.getName();
-      String albumDescription = cleanString(album.getDescription());
-
-      Photoset photoset = photosetsInterface.create(albumName, albumDescription, photoId);
-      monitor.debug(() -> String.format("%s: Flickr importer created album: %s", id, album));
-
-      // Update the temp mapping to reflect that we've created the album
-      tempData.addAlbumId(oldAlbumId, photoset.getId());
-      tempData.removeTempPhotoAlbum(oldAlbumId);
+      createAlbum(idempotentExecutor, jobId, oldAlbumId, photoId);
     } else {
       // We've already created the album this photo belongs in, simply add it to the new album
       photosetsInterface.addPhoto(newAlbumId, photoId);
     }
+  }
 
-    jobStore.update(id, createCacheKey(), tempData);
+  private void createAlbum(
```

Comment:
Not now but it might be nice to have this call a method creating the album without the idempotentExcecutor